File: src/main/kotlin/com/vk/axwer/kotlin/p511/Lambda.java
package com.vk.axwer.kotlin.p511;

import com.vk.axwer.kotlin.p512.Person;

import java.util.Arrays;
import java.util.Comparator;
import java.util.List;

public class Lambda {
    public static void main(String[] args) {
        List<Person> people = Arrays.asList(
                new Person("Vladimir", 30),
                new Person("Rinat", 27),
                new Person("Marat", 14),
                new Person("Dmitri", 28)
        );
        // Реализация компаратора с помощью анонимного внутреннего класса
        people.sort(new Comparator<Person>() {
            @Override
            public int compare(Person o1, Person o2) {
                return o1.getAge() - o2.getAge();
            }
        });
        System.out.println(people);
    }
}

File: src/main/kotlin/com/vk/axwer/kotlin/p511/lambda.kt
package com.vk.axwer.kotlin.p511

import com.vk.axwer.kotlin.p512.Person

// 5.1.1. Введение в лямбда выражения: фрагменты кода как параметры функций
fun main() {
    val people = listOf(
            Person("Vladimir", 30),
            Person("Rinat", 27),
            Person("Marat", 14),
            Person("Dmitri", 28)
    )
    // Использование лямбда-выражения
    people.sortedBy { it.age }
    println(people)
}
